### PR TITLE
fix(sync-bps): optimizing function performance

### LINKED ISSUE #549: Check and resolve console errors

### LINKED ISSUE #609: Change reduce function to map function
https://github.com/eoscostarica/eos-rate/blob/435c6c54880bc3f52f0a53d39991bfa78e24f806/services/hapi/src/libs/sync-bps.js#L22-L27

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a `TODO` comment in 435c6c54880bc3f52f0a53d39991bfa78e24f806 when #608 was merged. cc @leisterfrancisco.

### LINKED ISSUE #610: Optimize filter functions and if statements
https://github.com/eoscostarica/eos-rate/blob/435c6c54880bc3f52f0a53d39991bfa78e24f806/services/hapi/src/libs/sync-bps.js#L38-L41

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a `TODO` comment in 435c6c54880bc3f52f0a53d39991bfa78e24f806 when #608 was merged. cc @leisterfrancisco.

### LINKED ISSUE #611: Optimize filter functions and if statements
https://github.com/eoscostarica/eos-rate/blob/98fddc217b6171099dbf066f7a3ffcd792a88bac/services/hapi/src/libs/sync-bps.js#L38-L41

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a `TODO` comment in 98fddc217b6171099dbf066f7a3ffcd792a88bac. It&#x27;s been assigned to @leisterfrancisco because they committed the code.

### PR DESCRIPTION
**Pull request by leisterfrancisco**
### GH Issue

Resolves #549 
Resolves #609  
Resolves #610 
Resolves #611 

### Steps to test

1. Make run
1. Check console log

### Considerations
- Commit `17a7027` resolves sync-bps performance
- Commit `d782fe4` resolves sync-proxies performance
- Commit `94993ab` resolves sync-ratings performance
- Commit `746992e` resolves sync-stats performance
- Commit `15e54b8` resolves bpf-info performance

### CODE REVIEW

File: services/hapi/src/libs/sync-ratings.js [hunk 1]
```diff
@@ -20,66 +19,34 @@ const getUserRatings = async () => {
     reverse: false,
     show_payer: false
   })
```
Comment (leisterfrancisco) [inline services/hapi/src/libs/sync-ratings.js:21]: @xavier506 let me know if hard coded variables should be taken from .env variables, right?

  code: 'rateproducer',
  scope: 'rateproducer',

File: services/hapi/src/libs/add-bps-info.js [hunk 2]
```diff
@@ -365,30 +365,29 @@ const jungleBPinfo = [
   },
 ];
 
-const massive = require("massive");
+const massive = require("massive")
 const { massiveConfig } = require("../config");
```
Comment (leisterfrancisco) [inline services/hapi/src/libs/add-bps-info.js:369]: remove semicolon

**Review by xavier506 — APPROVED** [review]
LGTM

File: services/hapi/src/libs/sync-ratings.js (hunk 1 above)
Comment (xavier506) [reply services/hapi/src/libs/sync-ratings.js:21]: yes these should be sourced from env vars as they are different for jungle and mainnet environments

File: .env.example [hunk 3]
```diff
@@ -15,8 +15,12 @@ POSTGRES_DATA=./db_data
 HAPI_SERVER_PORT=9090
 HAPI_SERVER_ADDRESS=0.0.0.0
 EOS_API_ENDPOINT=https://jungle.eosio.cr
-PROXY_INFO_CONTRACT_CODE=proxyaccount
-PROXY_INFO_CONTRACT_SCOPE=proxyaccount
+HAPI_PROXY_CONTRACT_CODE=proxyaccount
+HAPI_PROXY_CONTRACT_SCOPE=proxyaccount
+HAPI_STATS_CONTRACT_CODE=rateproducer
```
Comment (xavier506) [inline .env.example:20]: @leisterfrancisco  

The values for  `HAPI_STATS_CONTRACT` and the `HAPI_RATING_CONTRACT` will always be the same 

I beleive we will be fine with just  `HAPI_RATING_CONTRACT`

**Review by AngeloCG97 — APPROVED** [review]

**Pull request merged**